### DISCUSSION TITLE
An error when calculating expect value of MCstate

### DISCUSSION THREAD

**Discussion by miaoL1** [netket/netket#1706] [General]
In a variational Monte Carlo algorithm, expectation is
<img width="356" alt="expectation" src="https://github.com/netket/netket/assets/82252965/4f24e470-e2d7-4e73-a5c1-35fd62742dc3">
where, O is operator, x belongs to "vstate.samples", {x'} is the complete set of Hilbert space, and \Psi is variational state.
The sign of <\Psi|x> and <\Psi|x'> do matter.
However, in "netket.vqs.mc.kernels.py", it calculates O(x) in such a following way:
```
@batch_discrete_kernel
def local_value_kernel(logpsi: Callable, pars: PyTree, σ: Array, args: PyTree):
    """
    local_value kernel for MCState and generic operators
    """
    σp, mel = args
    return jnp.sum(mel * jnp.exp(logpsi(pars, σp) - logpsi(pars, σ)))
```
jnp.exp(logpsi(pars, σp) - logpsi(pars, σ)) is always positive and, then gives the wrong result when  <\Psi|x'>/<\Psi|x> is negative, leading to false expectation.
I'm pondering how to fix this problem?

**Comment by gcarleo** [comment]
logpsi is complex in general, that quantity is not positive if your ansatz is in the complex domain

**Reply by miaoL1** [reply]
I see. Thanks
